test/jdk/javax/swing/JMenuItem/8031573/bug8031573.java

```diff
@@ -20,45 +20,105 @@
+            + "1. Run the test on HiDPI Display.\n"
+            + "2. Press the Menu in the applet.\n"
+            + "3. Check that the icon on the JCheckBoxMenuItem is smooth If so, press PASS, else press FAIL.\n";
 
-            SwingUtilities.invokeAndWait(new Runnable() {
+    public static void main(String args[]) throws Exception {
+        UIManager.setLookAndFeel(UIManager.getSystemLookAndFeelClassName());
+        SwingUtilities.invokeAndWait(() -> createTestGUI());
 
-                @Override
-                public void run() {
-                    JMenuBar bar = new JMenuBar();
-                    JMenu menu = new JMenu("Menu");
-                    JCheckBoxMenuItem checkBoxMenuItem
-                            = new JCheckBoxMenuItem("JCheckBoxMenuItem");
-                    checkBoxMenuItem.setSelected(true);
-                    menu.add(checkBoxMenuItem);
-                    bar.add(menu);
-                    setJMenuBar(bar);
-                }
-            });
-        } catch (Exception e) {
-            throw new RuntimeException(e);
+        if (!latch.await(5, TimeUnit.MINUTES)) {
+            frame.dispose();
+            throw new RuntimeException("Test has timed out!");
+        }
+        if (!passed) {
+            throw new RuntimeException("Test failed!");
         }
     }
+
+    private static void createTestGUI() {
+        frame = new JFrame("bug8031573");
+        frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+        JMenuBar bar = new JMenuBar();
+        JMenu menu = new JMenu("Menu");
+        JCheckBoxMenuItem checkBoxMenuItem = new JCheckBoxMenuItem("JCheckBoxMenuItem");
+        checkBoxMenuItem.setSelected(true);
+        menu.add(checkBoxMenuItem);
+        bar.add(menu);
+        frame.setJMenuBar(bar);
+
+        JPanel panel = new JPanel(new BorderLayout());
+        JTextComponent textComponent = new JTextArea(INSTRUCTIONS);
+        textComponent.setEditable(false);
+        panel.add(textComponent, BorderLayout.CENTER);
+
+        JPanel buttonsPanel = new JPanel(new FlowLayout());
+        JButton passButton = new JButton("Pass");
+        passButton.addActionListener((e) -> {
+            System.out.println("Test passed!");
+            passed = true;
+            latch.countDown();
+            frame.dispose();
+        });
+        JButton failsButton = new JButton("Fail");
+        failsButton.addActionListener((e) -> {
+            passed = false;
+            latch.countDown();
+            frame.dispose();
+            throw new RuntimeException("Test Failed!");
+        });
+
+        buttonsPanel.add(passButton);
+        buttonsPanel.add(failsButton);
+        panel.add(buttonsPanel, BorderLayout.SOUTH);
+
+        frame.getContentPane().add(panel);
+
+        frame.addWindowListener(new WindowAdapter() {
+            @Override
+            public void windowClosing(WindowEvent e) {
+                latch.countDown();
+            }
```

Comment:
Add here `frame.dispose();`  and then the frame will close without `frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);` and the app will exit even when launched as a stand-alone app.